File: src/EPI_Kotlin/epi/problem_5_8.kt
package epi

/**
 * Created by rick on 2017/1/16.
 */
fun reverseDigits(value: Long): Long {
    var result = 0L
    val negative = value < 0
    var absValue = Math.abs(value)

    while (absValue != 0L){
       val remainder = absValue % 10;
        result = result * 10 + remainder
        absValue /= 10
    }

    return if (negative) -result else result

}

File: src/EPI/EPI_TypeScript/src/problem_5_8.ts

export function ReverseDigits(value: number): number {
    let result = 0;
    let negtive = value < 0;
    value = Math.abs(value);

    while (value !== 0) {
        result = result * 10 + value % 10;
        value /= 10;
        value = Math.floor(value);
    }
    return negtive ? -result: result;
}